src/views/webviewController.ts

```diff
@@ -105,6 +106,54 @@ export default class WebviewController {
     this._themeChangedSubscription?.dispose();
   }
 
+  handleWebviewOpenFileChooserAttempt = async ({
+    panel,
+    fileChooserOptions,
+    requestId,
+  }: {
+    panel: vscode.WebviewPanel;
+    fileChooserOptions: FileChooserOptions;
+    requestId: string;
+  }): Promise<void> => {
+    let files;
+
+    try {
+      files = await vscode.window[
+        fileChooserOptions.mode === 'open' ? 'showOpenDialog' : 'showSaveDialog'
+      ]({
+        defaultUri: vscode.Uri.from({
+          path: fileChooserOptions.electronFileDialogOptions?.defaultPath,
+          scheme: 'file',
+        }),
+        [`${fileChooserOptions.mode}Label`]:
+          fileChooserOptions.electronFileDialogOptions?.buttonLabel,
+        filters: fileChooserOptions.electronFileDialogOptions?.filters?.reduce(
+          (acc, filter) => {
+            acc[filter.name] = filter.extensions;
+            return acc;
+          },
+          {}
+        ),
+        title: fileChooserOptions.electronFileDialogOptions?.title,
+      });
+    } catch (error) {
+      void vscode.window.showErrorMessage(
+        `Unable to open file chooser dialog: ${error}`
+      );
+    }
+
+    void panel.webview.postMessage({
+      command: MESSAGE_TYPES.OPEN_FILE_CHOOSER_RESULT,
+      fileChooserResult: {
+        canceled: false,
+        ...(fileChooserOptions.mode === 'open'
+          ? { filePaths: files ? files?.map((file) => file.path) : [] }
+          : { filePath: files?.path }),
+      },
+      requestId,
+    });
```

Comment:
This is getting executed, even if an exception occurs - is this what we want?

Reply:
Yes, we send the requestId back so we can remove the event listener and files will be undefined what is fine.